assign-doi-proxy-function/src/test/java/no/unit/nva/datacite/DataCiteMdsCreateDoiHandlerTest.java

```diff
@@ -135,7 +131,7 @@ public void handlerReturnsPaymentRequiredWhenInstitutionIdDataciteConfigNotFound
         InputStream input = requestWithBodyAndHeaders(MOCK_URL, MOCK_UNKNOWN_INSTITUTION_ID, MOCK_DATACITE_XML);
         dataCiteMdsCreateDoiHandler.handleRequest(input, output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GATEWAY_PROBLEM_TYPE);
```

Comment:
outputStream should be a ByteArrayOutputStream and you can use the function `GatewayResponse.fromOutputStream(outputStream)`

Example:
```
GatewayResponse<Problem> gatewayResponse = GatewayResponse.fromOutputStream(output);
```

Reply:
Fixed.